pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -113,36 +113,37 @@ func (pl *DefaultPreemption) preempt(ctx context.Context, state *framework.Cycle
 		return "", err
 	}
 
-	// 1) Calculate all Strategy candidates.
-	strategies, err := CalcStrategies(ctx, cs, state, pod, m, ph, nodeLister, pl.pdbLister)
-	if err != nil || len(strategies) == 0 {
+	// 1) Find all preemption candidates.
+	candidates, err := FindCandidates(ctx, cs, state, pod, m, ph, nodeLister, pl.pdbLister)
+	if err != nil || len(candidates) == 0 {
 		return "", err
 	}
 
-	// 2) Interact with registered Extenders.
-	strategies, err = CallExtenders(ph.Extenders(), pod, nodeLister, strategies)
+	// 2) Interact with registered Extenders to filter out some candidates if needed.
+	candidates, err = CallPreemptExtenders(ph.Extenders(), pod, nodeLister, candidates)
 	if err != nil {
 		return "", err
 	}
 
-	// 3) Find the best strategy.
-	strategy := PickStrategy(strategies)
-	if strategy == nil || len(strategy.NominatedNodeName()) == 0 {
+	// 3) Find the best candidate.
+	bestCandidate := SelectBestCandidate(candidates)
+	if bestCandidate == nil || len(bestCandidate.Name()) == 0 {
 		return "", nil
 	}
 
-	// 4) Execute the best-fit PreemptionStrategy.
-	if err := ExecuteStrategy(strategy, pl.fh, cs, pod); err != nil {
+	// 4) Apply the candidate winner.
+	if err := ApplyCandidate(bestCandidate, pl.fh, cs, pod); err != nil {
```

Comment:
actually lets name the function `NominateCandidate`

Reply:
Actually this function doesn't really "nominate" - which is done in the ErrorFunc. So how about `PreNominateCandidate`?